include/util/group_by.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef OSRM_GROUP_BY
+#define OSRM_GROUP_BY
+
+#include <algorithm>
+#include <utility>
+
+namespace osrm
+{
+namespace util
+{
+
+// Runs fn on consecutive items in sub-ranges determined by pred.
+//
+// Example:
+//   vector<int> v{1,2,2,2,3,4,4};
+//   group_by(first, last, even, print);
+//   >>> 2,2,2
+//   >>> 4,4
+template <typename Iter, typename Pred, typename Fn>
```

Comment:
For the record, this is basically
https://docs.python.org/3.5/library/itertools.html#itertools.groupby